core/templates/dev/head/services/RteHelperServiceSpec.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview unit tests for the Rich-text-editor Service.
+ */
+
+describe('RteHelperService', function() {
+  beforeEach(module('oppia'));
+  beforeEach(inject(function($injector) {
+    RteHelperService = $injector.get('RteHelperService');
+  }));
+
+  it('should return text with no extra new lines or spaces', function () {
+    expect(RteHelperService.convertRteToHtml(
+      '<p>abc</p>')).toEqual('<p>abc</p>');
+    expect(RteHelperService.convertRteToHtml('<p>abc</p><p><br></p><p>abc</p>'))
+      .toEqual('<p>abc</p><p><br></p><p>abc</p>');
+    expect(RteHelperService.convertRteToHtml(
+      '<p>abc</p><p><br></p><p>abc</p><p><br></p>'))
+      .toEqual('<p>abc</p><p><br></p><p>abc</p>');
+    expect(RteHelperService.convertRteToHtml(
+      '<p>abc</p><p><br></p><p>abc</p><p><br></p><p><br></p>'))
+      .toEqual('<p>abc</p><p><br></p><p>abc</p>');
+    expect(RteHelperService.convertRteToHtml('<p>abc</p><p><br><br></p>'))
+      .toEqual('<p>abc</p>');
+    expect(RteHelperService.convertRteToHtml('<p>abc</p><p>abc<br></p>'))
+      .toEqual('<p>abc</p><p>abc</p>');
+  });
```

Comment:
Good tests. What about `<p>abc</p><br><br>`?

Reply:
We actually don't need to worry about these cases. I looked into it and found that anything we type/insert into the text field gets wrapped around <p> tags, and that is ```rte``` that is passed into the function ```ConvertRteToHtml```. So if anything that is outside a <p> tag gets discarded automatically.

Reply:
I thought I explained this in a previous review -- the code should work in the general case, since it will persist through multiple implementations of the RTE. So we do need to worry about that and similar cases.

Reply:
In that case I'm updating the PR. Thanks!